File: python/test/test_types.py
from pydantic import Field

from json_tag import JsonTagModel, tagged_union


class Circle(JsonTagModel):
    radius: float


class Rectangle(JsonTagModel):
    width: float
    height: float


Shape = tagged_union(Circle, Rectangle)


class PersonalVehicle(JsonTagModel):
    __json_tag_name__ = "Car"
    make: str


class TransportationVehicle(JsonTagModel):
    __json_tag_name__ = "Truck"
    payload_tons: float = Field(alias="payloadTons")


Vehicle = tagged_union(PersonalVehicle, TransportationVehicle)

File: rust/json-tag-tests/tests/test_types.rs
use json_tag::json_tag;
use serde::{Deserialize, Serialize};

#[json_tag]
#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all_fields = "camelCase")]
pub enum Shape {
    Circle { radius: f64 },
    Rectangle { width: f64, height: f64 },
}

#[json_tag]
#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all_fields = "camelCase")]
pub enum Vehicle {
    #[serde(rename = "Car")]
    PersonalVehicle { make: String },
    #[serde(rename = "Truck")]
    TransportationVehicle { payload_tons: f64 },
}
